### ISSUE TITLE
CRI still forces https when an http endpoint is specified

### ISSUE THREAD

**Issue by joedborg** [containerd/cri#1433]
Not sure if this is linked to https://github.com/containerd/cri/issues/1367 and or https://github.com/containerd/cri/issues/1201 but I still get the error:

This is with containerd 1.3.3 and K8s 1.18.

```
[plugins."io.containerd.grpc.v1.cri".registry.mirrors."184.73.131.126:5000"]
  endpoint = ["http://184.73.131.126:5000"]
```

```
Events:
  Type     Reason     Age        From                       Message
  ----     ------     ----       ----                       -------
  Normal   Scheduled  <unknown>  default-scheduler          Successfully assigned default/test to ip-172-31-43-150
  Normal   Pulling    6s         kubelet, ip-172-31-43-150  Pulling image "184.73.131.126:5000/nginx:latest"
  Warning  Failed     6s         kubelet, ip-172-31-43-150  Failed to pull image "184.73.131.126:5000/nginx:latest": rpc error: code = Unknown desc = failed to pull and unpack image "184.73.131.126:5000/nginx:latest": failed to resolve reference "184.73.131.126:5000/nginx:latest": failed to do request: Head https://184.73.131.126:5000/v2/nginx/manifests/latest: http: server gave HTTP response to HTTPS client
  Warning  Failed     6s         kubelet, ip-172-31-43-150  Error: ErrImagePull
  Normal   BackOff    6s         kubelet, ip-172-31-43-150  Back-off pulling image "184.73.131.126:5000/nginx:latest"
  Warning  Failed     6s         kubelet, ip-172-31-43-150  Error: ImagePullBackOff

```

**Comment by solmon** [comment]
Hi Joe, I am stuck with this issue from past week badly.. is there any quick fix that is available? if not, any ETA on the release of this fix?
Thank you.

**Comment by joedborg** [comment]
Hi @solmon, I would also be very interested in a quick fix.  Sadly, I have not found one yet.

Not sure why this has been untouched for so long.

**Comment by solmon** [comment]
> Hi @solmon, I would also be very interested in a quick fix. Sadly, I have not found one yet.
> 
> Not sure why this has been untouched for so long.

Thank you Joe

**Comment by joedborg** [comment]
@mikebrow @Random-Liu, any chance you could have a look at this and let me know?  It's quite a blocker for us.

**Comment by mikebrow** [comment]
https://github.com/containerd/cri/commit/d07f7f167a437612b486a381addc0229b10d82bc
can you try master? see above fix..

**Comment by joedborg** [comment]
Thanks @mikebrow, I'll try that today.

**Comment by RussianNeuroMancer** [comment]
@joedborg is it works for you?

**Comment by balchua** [comment]
@joedborg do you happen to have the `tls` in the toml?
Like this?
`[plugins.cri.registry.configs."184.73.131.126:5000".tls]`

**Comment by joedborg** [comment]
hey @balchua!  I didn't, no.

**Comment by balchua** [comment]
Bummer. I got this one working sometime back. While working on upgrading containerd to 1.3.1 in microk8s.

**Comment by joedborg** [comment]
Hey all, 

Just tested this on the 1.4.0-beta and it's working!

**Comment by joedborg** [comment]
```toml
    [plugins.cri.registry]
      [plugins.cri.registry.mirrors]
        [plugins.cri.registry.mirrors."docker.io"]
          endpoint = ["https://registry-1.docker.io"]
        [plugins.cri.registry.mirrors."34.235.169.214:5000"]
          endpoint = ["http://34.235.169.214:5000"]
      [plugins.cri.registry.auths]
        [plugins.cri.registry.auths."34.235.169.214:5000"]
          username = "admin"
          password = "password"
```

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: test
  namespace: default
spec:
  containers:
  - name: test
    image: 34.235.169.214:5000/nginx:latest
```
```
Events:
  Type Reason Age From Message
  ---- ------ ---- ---- -------
  Normal Scheduled <unknown> default-scheduler Successfully assigned default/test to ip-172-31-17-195
  Normal Pulling 10m kubelet, ip-172-31-17-195 Pulling image "34.235.169.214:5000/nginx:latest"
  Normal Pulled 10m kubelet, ip-172-31-17-195 Successfully pulled image "34.235.169.214:5000/nginx:latest"
  Normal Created 10m kubelet, ip-172-31-17-195 Created container test
  Normal Started 10m kubelet, ip-172-31-17-195 Started container test
```

**Comment by joedborg** [comment]
This also appears fixed in 1.3.7.
